subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerJavaCachingSmokeTest.groovy

```diff
@@ -288,6 +297,271 @@ class AndroidSantaTrackerJavaCachingSmokeTest extends AbstractAndroidSantaTracke
+        ':santa-tracker:mergeDevelopmentDebugResources': SUCCESS,
+        ':santa-tracker:mergeDevelopmentDebugShaders': FROM_CACHE,
+        ':santa-tracker:mergeExtDexDevelopmentDebug': FROM_CACHE,
+        ':santa-tracker:mergeLibDexDevelopmentDebug': FROM_CACHE,
+        ':santa-tracker:mergeProjectDexDevelopmentDebug': FROM_CACHE,
+        ':santa-tracker:packageDevelopmentDebug': SUCCESS,
+        ':santa-tracker:preBuild': UP_TO_DATE,
+        ':santa-tracker:preDevelopmentDebugBuild': UP_TO_DATE,
+        ':santa-tracker:processDevelopmentDebugJavaRes': NO_SOURCE,
+        ':santa-tracker:processDevelopmentDebugManifest': FROM_CACHE,
+        ':santa-tracker:processDevelopmentDebugResources': SUCCESS,
+        ':santa-tracker:stripDevelopmentDebugDebugSymbols': NO_SOURCE,
+        ':santa-tracker:validateSigningDevelopmentDebug': FROM_CACHE,
+        ':village:assembleDebug': SUCCESS,
+        ':village:bundleDebugAar': SUCCESS,
+        ':village:bundleLibCompileToJarDebug': FROM_CACHE,
+        ':village:bundleLibResDebug': NO_SOURCE,
+        ':village:bundleLibRuntimeToJarDebug': FROM_CACHE,
+        ':village:compileDebugAidl': NO_SOURCE,
+        ':village:compileDebugJavaWithJavac': FROM_CACHE,
+        ':village:compileDebugLibraryResources': SUCCESS,
+        ':village:compileDebugRenderscript': NO_SOURCE,
+        ':village:compileDebugShaders': NO_SOURCE,
+        ':village:compileDebugSources': UP_TO_DATE,
+        ':village:copyDebugJniLibsProjectAndLocalJars': FROM_CACHE,
+        ':village:copyDebugJniLibsProjectOnly': FROM_CACHE,
+        ':village:extractDebugAnnotations': FROM_CACHE,
+        ':village:extractDeepLinksDebug': FROM_CACHE,
+        ':village:generateDebugAssets': UP_TO_DATE,
+        ':village:generateDebugBuildConfig': FROM_CACHE,
+        ':village:generateDebugRFile': FROM_CACHE,
+        ':village:generateDebugResValues': FROM_CACHE,
+        ':village:generateDebugResources': UP_TO_DATE,
+        ':village:javaPreCompileDebug': FROM_CACHE,
+        ':village:mergeDebugConsumerProguardFiles': FROM_CACHE,
+        ':village:mergeDebugGeneratedProguardFiles': FROM_CACHE,
+        ':village:mergeDebugJavaResource': FROM_CACHE,
+        ':village:mergeDebugJniLibFolders': FROM_CACHE,
+        ':village:mergeDebugNativeLibs': FROM_CACHE,
+        ':village:mergeDebugShaders': FROM_CACHE,
+        ':village:packageDebugAssets': FROM_CACHE,
+        ':village:packageDebugRenderscript': NO_SOURCE,
+        ':village:packageDebugResources': FROM_CACHE,
+        ':village:parseDebugLocalResources': FROM_CACHE,
+        ':village:preBuild': UP_TO_DATE,
+        ':village:preDebugBuild': UP_TO_DATE,
+        ':village:prepareLintJarForPublish': SUCCESS,
+        ':village:processDebugJavaRes': NO_SOURCE,
+        ':village:processDebugManifest': FROM_CACHE,
+        ':village:stripDebugDebugSymbols': NO_SOURCE,
+        ':village:syncDebugLibJars': FROM_CACHE,
+    ]
```

Comment:
Are the two versions significantly different? If not, I'd rather do several small ifs instead, so that we have a single list where the differences are highlighted.

Reply:
There are about a dozen differences.

Some are only about the outcome, `SUCCESS` or `FROM_CACHE`. There are also new tasks that didn't exist before. That could be handled with conditions but then it would defeat the intended simple usage of the fixture that outputs the Groovy code for expected outcomes. This whole thing is rolling, I expect we never will have more than 3 versions covered at a given time, most often 2. I'm balanced, I can consider the extra effort if you feel strongly about it.

Reply:
There's another reason: it is nice to be able to check back the history of each line and see which AGP versions they were changed for. But I'm not sure if it's so useful, and I don't feel very strongly.

If you keep the current setup I think we should rename `EXPECTED_RESULTS` to `EXPECTED_RESULTS_3_6` or something.

Reply:
I renamed `EXPECTED_RESULTS` to `EXPECTED_RESULTS_4_1`. Good point as it got you confused.